FABulous/FABulous_CLI/FABulous.py

```diff
@@ -0,0 +1,1100 @@
+            ]
+            if self.verbose or self.debug:
+                logger.info(f"Running simulation with {args.format} format")
+                logger.info(f"Running command: {' '.join(runCmd)}")
+            sp.run(runCmd, check=True)
+
+        except sp.CalledProcessError:
+            logger.error("Simulation failed")
+            remove_dir(tmpDir)
+            return
+
+        make_hex(bitstreamPath, bitstreamPath.with_suffix(".hex"))
+
+        vvp = check_if_application_exists(os.getenv("FAB_VVP_PATH", "vvp"))
+        try:
+            runCmd = [f"{vvp}", f"{bitstreamPath.parent}/{vvpFile}"]
+            sp.run(runCmd, check=True)
+        except sp.CalledProcessError:
+            logger.error("Simulation failed")
+            remove_dir(tmpDir)
+            return
+
+        remove_dir(tmpDir)
+        logger.info("Simulation finished")
+
+    @with_category(CMD_FABRIC_FLOW)
+    @with_argparser(filePathRequireParser)
+    def do_run_FABulous_bitstream(self, args):
+        """
+        Runs FABulous to generate bitstream on a given design starting from synthesis.
+
+        Does this by calling synthesis, place and route, bitstream generation functions.
+        Requires Verilog file specified by <top_module_file>.
+
+        Also logs usage error and file not found error.
+        """
+
+        file_path_no_suffix = args.file.parent / args.file.stem
+
+        if args.file.suffix != ".v":
+            logger.error(
+                """
+                No verilog file provided.
+                Usage: run_FABulous_bitstream <top_module_file>
+                """
+            )
+            return
+
+        json_file_path = file_path_no_suffix.with_suffix(".json")
+        fasm_file_path = file_path_no_suffix.with_suffix(".fasm")
+
+        self.do_synthesis(str(args.file))
+        self.do_place_and_route(str(json_file_path))
+        self.do_gen_bitStream_binary(str(fasm_file_path))
+
+    @with_category(CMD_SCRIPT)
+    @with_argparser(filePathRequireParser)
+    def do_run_tcl(self, args):
+        """Executes TCL script relative to the project directory, specified by
+        <tcl_scripts>. Uses the 'tk' module to create TCL commands.
+
+        Also logs usage errors and file not found errors.
+
+        Usage:
+            tcl <tcl_scripts>
+
+        Parameters
+        ----------
+        args : str
+            Path to the TCL script.
+        """
+        if not args.file.exists():
+            logger.error(f"Cannot find {args.file}")
+            return
+
+        logger.info(f"Execute TCL script {args.file}")
+        tcl = tk.Tcl()
+        for fun in dir(self.__class__):
+            if fun.startswith("do_"):
+                name = fun.strip("do_")
+                tcl.createcommand(name, getattr(self, fun))
+
+        tcl.evalfile(str(args.file))
+        logger.info("TCL script executed")
+
+
+def main():
+    """Main function to run command line interface for FABulous,
```

Comment:
Please update this comment, with the currently available command line arguments.